File: src/test/java/net/thimmwork/Partition.java
package net.thimmwork;

import com.google.common.base.Joiner;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Partition {
    private final String name;
    private final Object[] contents;


    public Partition(String name, Object... contents) {
        this.name = name;
        this.contents = contents;
    }

    @Override
    public String toString() {
        List<String> lines = new ArrayList<>(contents.length + 2);
        lines.add("partition " + name + " {");
        for (Object it : contents) {
            lines.add(it.toString());
        }
        lines.add("}");
        return Joiner.on("\n").join(lines);
    }
}

File: src/test/java/net/thimmwork/Partition.kt
package net.thimmwork

import com.google.common.base.Joiner

import java.util.ArrayList
import java.util.Arrays

class Partition(private val name: String, private vararg val contents: Any) {

    override fun toString(): String {
        val lines = ArrayList<String>(contents.size + 2)
        lines += "partition $name {"
        contents.forEach { lines += it.toString() }
        lines += "}"
        return Joiner.on("\n").join(lines)
    }
}
